### PR TITLE
Modify Airlines IA to use FlightStats v2 API

### PR DESCRIPTION
**Pull request by tommytommytommy**
This PR fixes the Airlines IA by making it explicitly use the FlightStats endpoint and brings us one step closer to unifying the Airlines and Flights/Route IAs!

### CODE REVIEW

File: lib/DDG/Spice/Airlines.pm [hunk 1]
```diff
@@ -49,25 +49,33 @@ foreach my $line (@elements_lines) {
...
     } else {
         return;
     }
 }
 
 handle query_lc => sub {
+
     my $query = $_;
 
+    # get the current time, minus six hours
```
Comment (jagtalon) [inline lib/DDG/Spice/Airlines.pm]: @tommytommytommy could you add information why it's minux 6 hours?

File: share/spice/airlines/airlines.js [hunk 2]
```diff
@@ -1,115 +1,255 @@
...
+            },
+            MONTH = [
+                "Jan.", "Feb.", "March", "April", "May", "June",
+                "July", "Aug.", "Sept.", "Oct.", "Nov.", "Dec." 
+            ];
 
         // Check if we have anything returned.
-        if (!api_result || !api_result.flight) {
-            return Spice.failed('airlines');
+        if (!api_result || !api_result.flightStatuses || !api_result.appendix.airlines) {
```
Comment (jagtalon) [inline share/spice/airlines/airlines.js]: Check if `api_result.appendix` exists first before accessing `airlines` property?

File: share/spice/airlines/airlines.js [hunk 3]
```diff
@@ -1,115 +1,255 @@
...
+                return hours + ":" + minutes + " " + suffix + " (" + MONTH[dateObject.getMonth()] + " " + dateObject.getDate() + ")";
+            } else
+                return hours + ":" + minutes + " " + suffix;
+        });
+
+        Spice.registerHelper("airline_status", function(flight, departureDate, arrivalDate, scheduledDeparture, scheduledArrival) {
+            var result = onTime(flight, departureDate, arrivalDate, scheduledDeparture, scheduledArrival),
+                ok_class = result[1] ? "tile__ok" : "tile__not";
+            return '<div class="' + ok_class + '">' + result[0] + '</div>';
+        });
```
Comment (jagtalon) [inline share/spice/airlines/airlines.js]: Nitpick: I'd like `relativeTime`, `onTime`, `airline_time`, and `airline_status` out of `ddg_spice_airlines` if possible just to be consistent with the other IAs.

File: share/spice/airlines/airlines.js [hunk 4]
```diff
@@ -1,115 +1,255 @@
...
-                    url: "http://www.flightstats.com/go/FlightStatus/flightStatusByFlight.do?&airlineCode=" +
-                        item.flight.Airline.AirlineCode +
-                        "&flightNumber=" +
-                        item.flight.FlightNumber
+                    url: "http://www.flightstats.com/go/FlightStatus/flightStatusByFlight.do?"
+                        + "airlineCode=" + item.airlineCode 
+                        + "&flightNumber=" + item.flightNumber
                 }
             },
             sort_fields: {
```
Comment (jagtalon) [inline share/spice/airlines/airlines.js]: Remove `sort_fields` because there's only one item?

Comment (jagtalon) [inline share/spice/airlines/airlines.js]: I was wrong--there could be multiple flights and it's good to sort them by date. 😬

**Comment by tommytommytommy** [comment]
@jagtalon Thanks for the comments! Made all the changes. Should be ready for testing.

**Comment by jagtalon** [comment]
@tommytommytommy looks like you need to disable the trigger for this one since we're using Deep to trigger it?

**Comment by jagtalon** [comment]
LGTM! :+1:

**Pull request merged**